### PR TITLE
Albert/hide layers

### LINKED ISSUE #307: Add the ability to have text layers in soccer default to hidden
When drawing shapes or text on the soccer field view, we pass a string indicating the "layer" that these show up on.  The "layers" tab has a check box for each layer to show or hide the content.  Right now every layer defaults to being enabled, which can lead to a lot of unneeded debug content being drawn that clutters up our pretty soccer field view.  It would be nice if we could add a way to hide some by default, then we can enable them when we're debugging specific things.

### PR DESCRIPTION
**Pull request by ashaw596**
Maybe what you wanted? It hides the motion planning stuff at least.
Closes #307

### CODE REVIEW

File: soccer/FieldView.hpp [hunk 1]
```diff
@@ -24,6 +24,11 @@ class FieldView : public QWidget
 			{
 				_layerVisible[i] = value;
 			}
+			else if (i >= _layerVisible.size())
```
Comment (justbuchanan) [inline soccer/FieldView.hpp]: What happens if `i` is greater than _layerVisible.size() by more than one?

File: soccer/MainWindow.cpp [hunk 2]
```diff
@@ -31,6 +31,8 @@ using namespace Eigen;
 QString LiveStyle("border:2px solid transparent");
 QString NonLiveStyle("border:2px solid red");
 
+std::vector<QString> hiddenLayers {"MotionControl", "Planning0", "Planning1", "Planning2", "Planning3", "Planning4", "Planning5", "time", "velocity"};
```
Comment (justbuchanan) [inline soccer/MainWindow.cpp]: This doesn't reflect the current visibility of layers in the UI, only the default/initial visibility.  Maybe rename it initiallyHiddenLayers?  Also, should probably be an ivar, not a global.

File: soccer/MainWindow.hpp [hunk 3]
```diff
@@ -71,6 +71,8 @@ class MainWindow : public QMainWindow
 
         QTimer updateTimer;
 
+		void addLayer(int i, QString name, bool checked);
```
Comment (justbuchanan) [inline soccer/MainWindow.hpp]: This should probably be private.

File: soccer/FieldView.hpp (hunk 1 above)
Comment (ashaw596) [inline soccer/FieldView.hpp]: Then it gets resized to that size and the intermediate values gets set whenever those i values get called. It should really never happen out of order though.

File: soccer/MainWindow.cpp (hunk 2 above)
Comment (ashaw596) [inline soccer/MainWindow.cpp]: What do you mean by an ivar?

**Comment by ashaw596** [comment]
So whatcha thing @justbuchanan

**Comment by justbuchanan** [comment]
Ok, looks good to me!

**Pull request merged**